ipi_tests/regression_tests/runstools.py

```diff
@@ -108,14 +117,30 @@ def _check_numpy_output(self, cwd):
                 test_output = np.loadtxt(self.tmp_dir / fname, usecols=self.usecol[ii])
 
                 try:
-                    np.testing.assert_allclose(
-                        test_output, ref_output, rtol=1.0e-7, atol=1.0e-15
-                    )
-                    # print("No anomaly during the regtest for {}".format(refname))
+                    if self.usetol[ii] is None:
+                        rtol, atol = (1e-7, 1e-15)
+                        np.testing.assert_allclose(
+                            test_output, ref_output, rtol=rtol, atol=atol
+                        )
+                    else:
+                        for icol, tol in enumerate(self.usetol[ii]):
+                            if tol is None:
+                                rtol, atol = (1e-7, 1e-15)
+                            else:
+                                rtol, atol = tol
+                            np.testing.assert_allclose(
+                                test_output[..., icol],
+                                ref_output[..., icol],
+                                rtol=rtol,
+                                atol=atol,
+                            )
                 except AssertionError:
                     raise AssertionError(
-                        "ANOMALY: Disagreement between reference and {} in {}".format(
-                            fname, str((self.parent / cwd).absolute())
+                        "ANOMALY: Disagreement between reference and {} in {}. Absolute discrepancy (mean,max): {},{}.".format(
```

Comment:
yeah would be nice but it'd require even more specialization. Perhaps the whole thing can be thought in a more compact way...
feel free to modify the logic to print an error per column.

Reply:
I see. I'm in Zeitnot with finishing my thesis, I don't think I can work on this until I submit. To avoid this PR to stale, I would say let's merge what is here.